### PR TITLE
Initial plugin implementation

### PR DESCRIPTION
**Pull request by yokuze**

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -2,32 +2,152 @@
...
    grunt.loadNpmTasks('grunt-eslint');
+   grunt.loadNpmTasks('grunt-sass');
+   grunt.loadNpmTasks('grunt-postcss');
+   grunt.loadNpmTasks('grunt-sass-lint');
 
-   grunt.registerTask('standards', [ 'eslint' ]);
+   grunt.registerTask('standards', [ 'eslint', 'sasslint' ]);
+   grunt.registerTask('build-js', [ 'browserify', 'uglify' ]);
+   grunt.registerTask('build-css', [ 'sass', 'postcss:styles' ]);
+   grunt.registerTask('build', [ 'build-js', 'build-css', 'copy:images' ]);
```
Comment (jthomerson) [inline Gruntfile.js:168]: Can we add a `develop` task that builds and watches for changes? It should basically cover the 95% usecase for what a developer wants to do when he's working on this repo.

File: package.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
   "name": "silvermine-videojs-airplay",
   "version": "0.9.0",
   "description": "video.js plugin for casting to airplay",
-  "main": "src/index.js",
+  "main": "src/js/index.js",
   "scripts": {
     "test": "./node_modules/.bin/istanbul cover ./node_modules/.bin/_mocha -- -R spec 'tests/**/*.test.js'"
```
Comment (jthomerson) [inline package.json]: Please add a `prepublish` script that builds the minified versions, as @onebytegone did at https://github.com/silvermine/videojs-quality-selector/pull/1/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R7

File: src/images/ic_airplay_black_24px.svg [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+        <path d="M0 0h24v24H0V0z" id="c"/>
+    </defs>
+    <clipPath id="b">
+        <use overflow="visible" xlink:href="#a"/>
+    </clipPath>
+    <clipPath clip-path="url(#b)" id="d">
+        <use overflow="visible" xlink:href="#c"/>
+    </clipPath>
+    <path clip-path="url(#d)" d="M6 22h12l-6-6zM21 3H3c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h4v-2H3V5h18v12h-4v2h4c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"/>
+</svg>
```
Comment (jthomerson) [inline src/images/ic_airplay_black_24px.svg]: Can you go ahead and fix the missing trailing newline on the SVGs that are missing it? Seems that one has it and two don't.

File: src/js/browser.js [hunk 4]
```diff
@@ -0,0 +1,6 @@
+'use strict';
```
Comment (jthomerson) [inline src/js/browser.js]: Please add a comment to this file indicating what it's used for.

File: src/js/enableAirPlay.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+function hasAirPlayAPISupport() {
+   return !!window.WebKitPlaybackTargetAvailabilityEvent;
+}
```
Comment (jthomerson) [inline src/js/enableAirPlay.js]: Why is this function duplicated in two files?

**Review by jthomerson — CHANGES_REQUESTED** [review]
Please also add some simple documentation in the form of an example. You can mimic what @onebytegone did in https://github.com/silvermine/videojs-quality-selector/pull/1/files since it's simple and accomplishes the purpose.

File: package.json [hunk 6]
```diff
@@ -25,19 +25,40 @@
...
+  "browserify": {
+    "transform": [
+      "browserify-shim"
+    ]
+  },
+  "browserify-shim": {
+    "video.js": "global:videojs"
+  },
+  "dependencies": {
+    "browserify-shim": "3.8.14"
```
Comment (jthomerson) [inline package.json]: Should this be a `devDependency`?

https://github.com/thlorenz/browserify-shim#you-will-always

**Comment by yokuze** [comment]
Alright @jthomerson, I think I've addressed your comments. Thanks.

**Comment by yokuze** [comment]
Sorry about that - had a whitespace issue that caused `eslint` to fail.

**Pull request merged**